rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1022,6 +1022,32 @@ final class SingletonContentSet extends ContentSet, TSingletonContentSet {
   override Content getAReadContent() { result = c }
 }
 
+final class OptionalStep extends ContentSet, TOptionalStep {
```

Comment:
This badly needs a QLDoc.  As it is one might assume it's something to do with Rusts `Option<T>`.

Also, as I understand it these contents are special in that they're never actually put on an access path during data flow, just used to label certain kinds of taint steps.  If that's accurate I think this fact should be called out as well.

Reply:
That's absolutely right. This is still a bit of a work-in-progress attempt to integrate @asgerf 's ideas from https://github.com/github/codeql/pull/17412 into our library.

Reply:
Looks like this has been addressed, thanks.